lib/inspec/secrets/yaml.rb

```diff
@@ -24,11 +24,17 @@ def initialize(target)
         @inputs = ::YAML.load_file(target)
       end
 
-      if @inputs == false || !@inputs.is_a?(Hash)
-        Inspec::Log.warn("#{self.class} unable to parse #{target}: invalid YAML or contents is not a Hash")
-        @inputs = nil
+      # In case of empty yaml file raise the warning else raise the parsing error.
+      if @inputs.nil? || @inputs == false
```

Comment:
for this case, shouldn't be `.empty?`

Reply:
> why would `@inputs` be false?

As per my knowledge, YAML load_file used to return false in case of empty files so I thought to keep both the checks in place. Currently, it returns nil. ( this is the only reference I found https://github.com/ruby/psych/issues/149)

Reply:
in ruby 2.7.0 it returns false
```
2.7.0 :002 > inputs = ::YAML.load_file('/workspace/profiles/ssh-baseline/files/empty-waivers1.yaml')
2.7.0 :003 > inputs.nil?
 => false 
2.7.0 :004 > inputs.class
 => FalseClass 
```

Reply:
```suggestion
      if !@inputs || @inputs.empty?
```